nodejs/week3/homework/src/backend/api/meals.js

```diff
@@ -0,0 +1,209 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+
+
+
+// api/meals/	GET	Returns all meals	GET api/meals/
+
+router.get("/", async(request, response) => {
+    try {
+        const { maxPrice, availableReservations, title, createdAfter, limit, } = request.query;
+        const onlyMeals = await knex("meal").select("*");
+        // Get meals that has a price smaller than maxPrice
+        if (maxPrice) {
+            try {
+                const mealsLessThanMaxPrice = await knex("meal")
+                    .where("price", "<", maxPrice)
+                    .then((data) => {
+                        response.json(data);
+                    });
+            } catch (error) {
+                throw error;
+            }
+            // Get meals that still has available reservations
+        } else if (availableReservations === "true") {
+            try {
+                const mealsFromDb = await knex("meal").select("*")
```

Comment:
I recommend experimenting with the following here: 
```
const mealsFromDb = await knex("meal")
          .select("*")
          .count("reservation.id as count")
          .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
          .groupBy("meal.id")
          .having(
            knex.raw("meal.max_reservations > count(reservation.meal_id)")
          );
```

And then of course remember to send the result as a response.